nornir/core/deserializer/configuration.py

```diff
@@ -0,0 +1,184 @@
+import importlib
+import logging
+from typing import Any, Callable, Dict, List, Optional, Type, cast
+
+from nornir.core import configuration
+from nornir.core.inventory import Inventory
+
+from pydantic import BaseSettings, Schema
+
+import ruamel.yaml
+
+
+logger = logging.getLogger("nornir")
+
+
+class SSHConfig(BaseSettings):
+    config_file: str = Schema(
+        default="~/.ssh/config", description="Path to ssh configuration file"
+    )
+
+    class Config:
+        env_prefix = "NORNIR_SSH_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(self, **kwargs) -> configuration.SSHConfig:
+        s = SSHConfig(**kwargs)
+        return configuration.SSHConfig(**s.dict())
+
```

Comment:
I propose `cls` instead of `self` to follow convention and `s = cls(**kwargs)`
The same would apply to other classes below

Reply:
done